src/cpuid/src/amd/mod.rs

```diff
@@ -0,0 +1,301 @@
+                    _ => (),
+                }
+            }
+        }
+
+        // Update extended apic id entry
+        {
+            // When hyper-threading is enabled each pair of 2 consecutive logical CPUs
+            // will have the same core id since they represent 2 threads in the same core.
+            // For Example:
+            // logical CPU 0 -> core id: 0
+            // logical CPU 1 -> core id: 0
+            // logical CPU 2 -> core id: 1
+            // logical CPU 3 -> core id: 1
+            let core_id =
+                // SAFETY: We know `cpus_per_core != 0` therefore this is always safe.
+                unsafe { u32::from(cpu_index.checked_div(cpus_per_core).unwrap_unchecked()) };
+
+            let leaf_8000001e = self
+                .leaf_mut::<0x8000001e>()
+                .ok_or(ExtendedApicIdError::MissingLeaf0x8000001e)?;
+            leaf_8000001e
+                .eax
+                .extended_apic_id_mut()
+                .checked_assign(u32::from(cpu_index))
+                .map_err(ExtendedApicIdError::ExtendedApicId)?;
+
+            leaf_8000001e
+                .ebx
+                .compute_unit_id_mut()
+                .checked_assign(core_id)
+                .map_err(ExtendedApicIdError::ComputeUnitId)?;
+            leaf_8000001e
+                .ebx
+                .threads_per_compute_unit_mut()
+                // SAFETY: We know `cpus_per_core > 0` therefore this is always safe.
+                .checked_assign(u32::from(unsafe {
+                    cpus_per_core.checked_sub(1).unwrap_unchecked()
+                }))
+                .map_err(ExtendedApicIdError::ThreadPerComputeUnit)?;
+
+            // SAFETY: We know the value always fits within the range and thus is always safe.
+            unsafe {
+                // This value means there is 1 node per processor.
+                leaf_8000001e
+                    .ecx
+                    .nodes_per_processor_mut()
+                    .unchecked_assign(0);
+                // Put all the cpus in the same node.
+                leaf_8000001e.ecx.node_id_mut().unchecked_assign(0);
+            }
+        }
+
+        // Update brand string entry
+        self.apply_brand_string(Self::DEFUALT_BRAND_STRING)
+            .map_err(NormalizeCpuidError::BrandString)?;
+
+        Ok(())
+    }
+}
+
+/// Error type for [`<AmdCpuidNotSupported as Supports>::supports`].
+#[derive(Debug, thiserror::Error, Eq, PartialEq)]
+#[error("AmdCpuidNotSupported.")]
+pub struct AmdCpuidNotSupported;
+
+impl Supports for AmdCpuid {
+    type Error = AmdCpuidNotSupported;
+    /// Checks if `self` is a able to support `other`.
+    ///
+    /// Checks if a process from an environment with CPUID `other` could be continued in an
+    /// environment with the CPUID `self`.
+    #[inline]
+    fn supports(&self, _other: &Self) -> Result<(), Self::Error> {
+        Ok(())
```

Comment:
Maybe add a comment why this always succeeds?

Reply:
Added a comment:
```
// We don't currently implement support checking for AMD, therefore we allow all templates, presuming support.
```